feat: add ci blog post (#334)
shorebirdtech/website: src/content/blog/introducing-shorebird-ci.md

## Why We Built Shorebird CI

From countless conversations with customers, we kept hearing the same story.
Code Push made deploying instant updates effortless, but their CI pipelines were
still slow, clunky, and expensive. Some teams even had entire roles dedicated to
maintaining CI.

That’s not how it should be. CI should be fast, invisible, and effortless. You
should be spending your time building features and customer value, not
pipelines.

## Our Focus

In building Shorebird CI we focused on 4 key points.

- **Blazingly Fast** — In our tests, Shorebird CI delivered up to 5× faster
  pipelines compared to typical setups. Thanks to quick startup times and
  reliable change detection, we run only the affected packages and automatically
  parallelize builds for maximum speed.
- **Zero-Config Setup** - Forget giant YAML files and bash scripts. By
  statically analyzing your project, Shorebird CI can infer most of your config
  automatically.

> - **Dart-Native** - Works with any Dart or Flutter project. Always up to date

Reviewer: ```suggestion
- **Dart-Native** - Works with any Dart or Flutter project. It's always up to date
```

  with the latest Dart and Flutter changes so you’re never blocked from adopting
  new features.

> - **Mono-repo Aware** - Growing into a mono-repo? We handle it out of the box,

Reviewer: We should update cspell then. "monorepo" is one word https://en.wikipedia.org/wiki/Monorepo

  so you can start small and scale without painful migrations.

From indie apps to Fortune 500 enterprises moving to Flutter, these principles
make CI faster, simpler, and more reliable for everyone.

## Pricing

Much like our pricing for Code Push, we want to be able to serve the community
and open source needs while allowing for larger customers with more extensive
needs. While we are in this Beta period we have 2 tiers.

- Hobby - Free during this beta period for public repositories. This includes
  all Shorebird CI features so you can run fast, zero-config pipelines for your
  community projects.
- Pro — A simple $20 per month. This includes up to 100 hours of compute per
  month in a secure, private tenant to keep your code safe. This plan is
  required for use with private repositories.

Pricing may change after this beta period as we learn more about customer usage
and move toward general availability. Our goal will always be simple,
transparent, and scale with your need pricing.

## Getting Started

Zero-config really means zero-config. With just a few quick steps, you’ll have
your account linked and be up and running.

> 1. Log into the [Shorebird Web Console](https://console.shorebird.dev) and and

Author: Damn it.

   select CI from the navigation panel.
2. Sign in with your GitHub account to grant access to your repositories.

> 3. In the Web Console, enable access for your organization and choose the

Reviewer: ```suggestion
3. In the Web Console, enable access for your GitHub organization and choose the
```

---
title: Introducing Shorebird CI
author: shorebirdtech
description:
  Announcing a new service from Shorebird, zero config CI for Flutter and Dart
  applications.
date: 2025-08-15

> cover: '1.0-cover.png'

Author: Yeah I just copied over something so it would work. I'm looking for a new icon right now.

> cover: '1.0-cover.png'

Author: Fixed.

fiddling with CI pipelines that weren’t built with Flutter in mind? Why isn’t
there a CI solution that just works out of the box and enforces best practices?

There wasn’t one, so we built it.

<iframe width="560" height="315" src="https://www.youtube.com/embed/ZMMV418Dt80?si=Ec4cphd4vm1WFBEI" style="display:block;margin: 0 auto;" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="strict-origin-when-cross-origin" allowfullscreen></iframe>

## Why We Built Shorebird CI

From countless conversations with customers, we kept hearing the same story.
Code Push made deploying instant updates effortless, but their CI pipelines were
still slow, clunky, and expensive. Some teams even had entire roles dedicated to
maintaining CI.

That’s not how it should be. CI should be fast, invisible, and effortless. You
should be spending your time building features and customer value, not
pipelines.

## Our Focus

In building Shorebird CI we focused on 5 key points.

- **Blazingly Fast** - In our tests, moving to Shorebird CI delivered up to 5×

>   faster pipelines compared to typical Flutter builds.

Reviewer 2: Might want to focus on how our runners have very low cold start times and checks are run in parallel whenever possible

> - **Zero-Config Setup** - Forget giant YAML files. By building directly on top
>   of the Flutter framework, we can infer most of your config automatically.

Reviewer 2: ```suggestion
- **Zero-Config Setup** - Forget giant YAML files and bash scripts. By statically analyzing your project, Shorebird CI can infer most of your config automatically.
```

- **Dart-Native** - Works with any Dart or Flutter project. Always up to date
  with the latest Dart and Flutter changes so you’re never blocked from adopting
  new features.
- **Mono-repo Aware** - Growing into a mono-repo? We handle it out of the box,
  so you can start small and scale without painful migrations.
- **Truly Multi-Platform** - Build for any platform Flutter supports. Your
  customers aren’t on one device, and your CI shouldn’t be either.

From indie apps to Fortune 500 enterprises moving to Flutter, these principles
make CI faster, simpler, and more reliable for everyone.

## Pricing

- Open Source — Free to use for public repos.

> - Private Repos — $20/month for all features, plus a secure, private tenant to

Reviewer 2: Is it $20/month for unlimited builds? Might be worth being explicit about whether there's a usage limit.

Reviewer 3: Let’s start with 100 hrs of compute per month.

  keep your code safe.

Simple. Transparent. Scales with you.

## Getting Started

Zero-config really means zero-config. You’ll be up and running in three quick
steps:

1. Install the [Shorebird CI GitHub app](https://github.com/apps/shorebird-ci).
2. Select the repositories you'd like to connect.
3. Open a pull request.

Yep, that's really it.

SCREENSHOT GOES HERE

From there, Shorebird CI automatically runs checks for:

- Formatting
- Analyzer

> - Unit tests

Reviewer 2: ```suggestion
- Unit tests
- Widget tests
```

---
title: Introducing Shorebird CI
author: shorebirdtech
description:
  Announcing a new service from Shorebird, zero config CI for Flutter and Dart
  applications.
date: 2025-08-15
cover: 'introducing-shorebird-ci-cover.png'
---

We’re thrilled to announce our newest product:
[Shorebird CI](https://ci.shorebird.dev) — a zero-config continuous integration
system built exclusively for Flutter and Dart by the very people who created
Flutter.

You’ve got an app to ship. Who wants to wrestle with YAML? Or spend hours
fiddling with CI pipelines that weren’t built with Flutter in mind? Why isn’t
there a CI solution that just works out of the box and enforces best practices?

> There wasn’t one, so we built it.

Author: Added.

Code Push made deploying instant updates effortless, but their CI pipelines were
still slow, clunky, and expensive. Some teams even had entire roles dedicated to
maintaining CI.

That’s not how it should be. CI should be fast, invisible, and effortless. You
should be spending your time building features and customer value, not
pipelines.

## Our Focus

In building Shorebird CI we focused on 5 key points.

- **Blazingly Fast** - In our tests, moving to Shorebird CI delivered up to 5×
  faster pipelines compared to typical pipelines. This is based off of our quick
  startup times and automatic parallelization.
- **Zero-Config Setup** - Forget giant YAML files and bash scripts. By
  statically analyzing your project, Shorebird CI can infer most of your config
  automatically.
- **Dart-Native** - Works with any Dart or Flutter project. Always up to date
  with the latest Dart and Flutter changes so you’re never blocked from adopting
  new features.
- **Mono-repo Aware** - Growing into a mono-repo? We handle it out of the box,
  so you can start small and scale without painful migrations.
- **Truly Multi-Platform** - Build for any platform Flutter supports. Your

>   customers aren’t on one device, and your CI shouldn’t be either.

Reviewer 3: This isn't true for this CI release.  When we do CD we will, for now we only run CI on linux.

Author: Will remove.

From indie apps to Fortune 500 enterprises moving to Flutter, these principles
make CI faster, simpler, and more reliable for everyone.

## Pricing

- Open Source — Free to use for public repos.
- Private Repos — $20/month for 100 hours of compute per month, plus a secure,

>   private tenant to keep your code safe.

Reviewer 3: $20/mo is beta pricing (that they could "lock in now"?), we're likely going to change it once we know more about our costs.  We offer up to 100 hours of compute per month at $20/mo, call us for more.

Simple. Transparent. Scales with you.

## Getting Started

Zero-config really means zero-config. You’ll be up and running in three quick
steps:

1. Install the [Shorebird CI GitHub app](https://github.com/apps/shorebird-ci).
2. Select the repositories you'd like to connect.
3. Open a pull request.

Yep, that's really it.

![Showing Shorebird CI in GitHub Checks in a PR](../../assets/blog/introducing-shorebird-ci/shorebird_ci_in_github_checks.png)

From there, Shorebird CI automatically runs checks for:

> - Formatting

Reviewer 3: Dart Code Formatting?

> - Analyzer

Reviewer 3: Dart Analyzer?
